.circleci/config.yml

```diff
@@ -661,6 +660,8 @@ workflows:
             - halt-puppeteer-check
           requires:
             - puppeteer-env-setup
+            # PR commits e2e tests waits for workflow triggered by master branch merge to finish.
+            - wait_until_previous_workflow_done
```

Comment:
I think we ought to block PR e2e testing while there is workflow running on master branch. I have seen some test failures caused by ui and api deploy jobs. What do you think?